relay_test.go

```diff
@@ -25,18 +25,18 @@ func serviceNameOpts(s string) *testutils.ChannelOpts {
 	return testutils.NewOpts().SetServiceName(s)
 }
 
-func withRelayedEcho(t testing.TB, f func(relay, server, client *Channel)) {
+func withRelayedEcho(t testing.TB, f func(relay, server, client *Channel, ts *testutils.TestServer)) {
```

Comment:
nit: can you break out a separate PR where this is changed to instead be `f func(ts *testutils.TestServer, client *Channel)` and the below tests just create locals for the `relay` and `server`?